src/search/Search.tsx

```diff
@@ -1,53 +1,51 @@
 import { useState, useEffect, useRef, useCallback } from "react";
 import { Action } from "../core/Action";
-import { SearchController } from "./SearchController";
 
 export interface SearchState {
     selectedIndex: number,
     query: string
 }
 
-function selectNext(state: SearchState, results: Action[]): SearchState {
-    var index = state.selectedIndex + 1;
-    if (index >= results.length) {
-        index = results.length-1;
-    }
-    return {...state, selectedIndex: index};
-}
-
-function selectPrev(state: SearchState): SearchState {
-    var index = state.selectedIndex - 1;
-    if (index < 0) {
-        index = 0;
-    }
-    return {...state, selectedIndex: index};
-}
-
 interface SearchProps {
-    onClose?: () => void,
-    controller: SearchController
+    initialQuery: string,
+    results: Action[],
+    prompt: string,
+    onCloseRequest: () => void,
+    onSearch: (query: string) => void,
 }
 
-export function Search({ controller, onClose }: SearchProps) {
-    const _onClose = onClose || (() => { controller.hideSearch() });
-    const _onSearch = (query: string) => { controller.search(query) };
-    const _initialQuery = controller.initialQuery;
-    const results = controller.searchResults;
-    const visible = controller.searchVisible;
+export function Search({ 
```

Comment:
Pewnie najlepiej byłoby gdyby Search był swoim własnym komponentem :)

Reply:
Tak, ogarnę to sobie jak będę faktycznie chciał tego użyć. Na razie to była tylko próba przeniesienia i nauczenia się jak to robić (i jak nie robić) ;)